### PR TITLE
DB Performance Optimization

### PR DESCRIPTION
**Pull request by harry1453**
Benchmark:

CPU: Core i7-8550U
Database: MariaDB 10.3 atop NVMe SSD
Time taken to sync to block 600k from genesis: 7 hours 16 minutes

Don't have a benchmark from before because it took ~a week to sync and I can't be bothered to time that

### CODE REVIEW

File: src/brs/BlockchainProcessorImpl.java [hunk 1]
```diff
@@ -948,17 +947,6 @@ private void pushBlock(final Block block) throws BlockNotAcceptedException {
...
-        ArrayList<Long> accountIds = new ArrayList<>();
-        block.getTransactions().forEach(t -> {
-          if (t.getRecipientId() != 0L)
-            accountIds.add(t.getRecipientId());
-          if (t.getSenderId() != 0L)
-            accountIds.add(t.getSenderId());
-        });
-        if (!accountIds.isEmpty()) {
-          stores.getAccountStore().getAccountTable().fillCache(accountIds);
-        }
```
Comment (blankey1337) [inline src/brs/BlockchainProcessorImpl.java]: we can just lose this, no issues?

File: src/brs/BlockchainProcessorImpl.java [hunk 2]
```diff
@@ -859,9 +859,8 @@ void setGetMoreBlocks(boolean getMoreBlocks) {
   }
 
   private void addBlock(Block block) {
-    if (blockchainStore.addBlock(block)) {
-      blockchain.setLastBlock(block);
-    }
+    blockchainStore.addBlock(block);
+    blockchain.setLastBlock(block);
```
Comment (blankey1337) [inline src/brs/BlockchainProcessorImpl.java]: minor logic tweak here, all good?

File: src/brs/db/cache/DBCacheManagerImpl.java [hunk 3]
```diff
@@ -43,22 +43,20 @@ public void close() {
...
-    CacheConfiguration cacheConfiguration = caches.get(name);
-    return cacheManager.getCache(name, cacheConfiguration.getKeyType(), cacheConfiguration.getValueType());
+  private <V> Cache<BurstKey, V> getEHCache(String name, Class<V> valueClass) {
+    return cacheManager.getCache(name, BurstKey.class, valueClass);
   }
 
-  public Cache getCache(String name) {
-    Cache cache = getEHCache(name);
-    return statisticsEnabled ? new StatisticsCache(cache, name, statisticsManager) : cache;
+  public <V> Cache<BurstKey, V> getCache(String name, Class<V> valueClass) {
```
Comment (blankey1337) [inline src/brs/db/cache/DBCacheManagerImpl.java]: nice job typing

File: src/brs/BlockchainProcessorImpl.java (hunk 1 above)
Comment (harry1453) [reply src/brs/BlockchainProcessorImpl.java]: The way this works is tries to predict what account ids will be needed and load them into cache ahead of time. However, this is guaranteed to be slower, even with perfect prediction (which it doesn't have), than caching on demand (when you retrieve the account)

Example: You need to use 2 accounts 2 times each. You cache them ahead of time and then each time you use them get them from the cache. You have done 2 database hits and 4 cache hits.

Now say we don't cache ahead of time and cache on demand. The first time you try to get each account they are not cached, so you get from db and store in cache. The second time you just hit the cache. You have done 2 database hits and 2 cache hits.

This alone means it's unnecessary, not even adding that the caching prediction isn't exactly perfect.

File: src/brs/BlockchainProcessorImpl.java (hunk 2 above)
Comment (harry1453) [reply src/brs/BlockchainProcessorImpl.java]: `blockchainStore.addBlock()` will always return true. so yes

File: src/brs/db/cache/DBCacheManagerImpl.java (hunk 3 above)
Comment (harry1453) [reply src/brs/db/cache/DBCacheManagerImpl.java]: thanks :)

**Pull request merged**